### ISSUE TITLE
[UI] Publish to Meshery Catalog Action should not be available while using Local Provider

### ISSUE THREAD

**Issue by leecalcote** [meshery/meshery#7102]
#### Current Behavior
Publishing to Meshery Catalog isn't a capability of the Local Provider. 

#### Desired Behavior
When users are signed in under the "None" (Local) Provider, the "Publish" button should not be displayed in the Designs, Filters, or Applications pages.

#### Screenshots/Logs
<img width="640" alt="Screenshot 2023-02-16 at 9 23 11 AM" src="https://user-images.githubusercontent.com/7570704/219410192-e775baab-edd7-42d6-9228-693c25f03c2f.png">

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

**Comment by krishna9304** [comment]
Please assign me this issue. @leecalcote

**Comment by leecalcote** [comment]
Sounds good, @krishna9304

**Comment by leecalcote** [comment]
Ultimately, we'll want to clean up this modal a bit as well and ensure that it is reusing the same modal component and same styles on input fields, too.

**Comment by krishna9304** [comment]
> Ultimately, we'll want to clean up this modal a bit as well and ensure that it is reusing the same modal component and same styles on input fields, too.

Great, I'll have a look into it and make sure everything is as per expectation of this issue.

**Comment by leecalcote** [comment]
@krishna9304 Thanks for jumping on this issue. Have you been able to build Meshery UI?

**Comment by krishna9304** [comment]
> @krishna9304 Thanks for jumping on this issue. Have you been able to build Meshery UI?

@leecalcote Yess, I built the meshery UI, I was exploring the codebase and found that one possible solution to solve this issue would be to get the globally logged in user and check if it is the local provider we can prevent the publish functionality.

One issue that I'm running into is, I'm not sure whether to get the details of the logged in user from the **redux** store or from the GET `/api/user` route?

Second issue that I'm facing is, I'm not sure how would I enable live reload in meshery UI. I followed the steps in the `ui/README.md` file but none of them seems to work for me.
